### PR TITLE
Allow templating for alertmanager

### PR DESCRIPTION
**Pull request by atopuzov**
User can now specify custom templates (as described in [1]) in  `templateFiles` in `values.yaml`.
Eg.

```
templateFiles:
  template_1.tmpl: |-
    .... template file
```
Also has to add:
```
    templates:
    - '*.tmpl'
```
in `config` (the config that populates `alertmanager.yaml`).

[1] https://github.com/coreos/prometheus-operator/blob/master/Documentation/user-guides/alerting.md

### CODE REVIEW

File: helm/alertmanager/values.yaml [hunk 1]
```diff
@@ -17,6 +17,10 @@ config:
     receiver: webhook
     repeat_interval: 12h
 
+## Alertmanager template files to include
+#
+templateFiles: {}
+
```
Comment (gianrubio) [inline helm/alertmanager/values.yaml]: Could you provide an example keeping the content commented?

Comment (atopuzov) [reply helm/alertmanager/values.yaml]: I use something like this, shamelessly stolen from [1]. I'll add a more simpler example to the code.
```
  templateFiles:
    template_1.tmpl: |-
      {{ define "cluster" }}{{ .ExternalURL | reReplaceAll ".*alertmanager\\.(.*)" "$1" }}{{ end }}

      {{ define "slack.myorg.text" }}
      {{- $root := . -}}
      {{ range .Alerts }}
        *Alert:* {{ .Annotations.summary }} - `{{ .Labels.severity }}`
        *Cluster:*  {{ template "cluster" $root }}
        *Description:* {{ .Annotations.description }}
        *Graph:* <{{ .GeneratorURL }}|:chart_with_upwards_trend:>
        *Runbook:* <{{ .Annotations.runbook }}|:spiral_note_pad:>
        *Details:*
          {{ range .Labels.SortedPairs }} • *{{ .Name }}:* `{{ .Value }}`
          {{ end }}
      {{ end }}
      {{ end }}
```
[1] https://engineering.infinityworks.com/slack-prometheus-alertmanager/

Comment (atopuzov) [reply helm/alertmanager/values.yaml]: This si what it looks like:
template

**Comment by gianrubio** [comment]
lgtm

**Comment by brancz** [comment]
thanks!

**Pull request merged**